streamingccl: flake in TestStreamPartition/stream-table-cursor-error

Found here on an unrelated PR: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/10289209?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
No pass/skip/fail event found for test
=== RUN   TestStreamPartition/stream-table-cursor-error
panic: test timed out after 14m55s

goroutine 61740 [running]:
testing.(*M).startAlarm.func1()
  GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
  GOROOT/src/time/sleep.go:176 +0x32
goroutine 1 [chan receive, 14 minutes]:
testing.(*T).Run(0xc003c83860, {0x60e2506?, 0x656ae5?}, 0x645ee28)
  GOROOT/src/testing/testing.go:1494 +0x37a
testing.runTests.func1(0xc00133c1e0?)
  GOROOT/src/testing/testing.go:1846 +0x6e
testing.tRunner(0xc003c83860, 0xc00373fc08)
  GOROOT/src/testing/testing.go:1446 +0x10b
testing.runTests(0xc003dfadc0?, {0xa9884a0, 0x7, 0x7}, {0x0?, 0x0?, 0xae72260?})
  GOROOT/src/testing/testing.go:1844 +0x456
testing.(*M).Run(0xc003dfadc0)
  GOROOT/src/testing/testing.go:1726 +0x5d9
github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamproducer_test.TestMain(0x613e7ca?)
  github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamproducer_test/pkg/ccl/streamingccl/streamproducer/main_test.go:31 +0x165
main.main()
  main/bazel-out/k8-fastbuild/bin/pkg/ccl/streamingccl/streamproducer/streamproducer_test_/testmain.go:121 +0x472


goroutine 19344 [select, 8 minutes]:
github.com/cockroachdb/cockroach/pkg/rpc.(*pipe).recv(0xc0018e5000, {0x76d6b60, 0xc0007acb40})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1434 +0xe8
github.com/cockroachdb/cockroach/pkg/rpc.pipeReader.recv({0xc0018e5000?, 0x64600a8?}, {0x76d6b60?, 0xc0007acb40?}, {0x5ea97e0, 0xc0073aa780})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1470 +0x45
github.com/cockroachdb/cockroach/pkg/rpc.serverStream.RecvMsg(...)
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1548
github.com/cockroachdb/cockroach/pkg/rpc.muxRangeFeedServerAdapter.Recv({{0x770eb30?, 0xc00a9d2f20?}})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1382 +0x59
github.com/cockroachdb/cockroach/pkg/server.(*Node).MuxRangeFeed(0xc00427e000, {0x7723ad0?, 0xc0018e50f0})
  github.com/cockroachdb/cockroach/pkg/server/node.go:1619 +0x138
github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.MuxRangeFeed.func3.1({0x0?, 0x76d6b60?}, {0x770eb30?, 0xc00a9d2f20?})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1309 +0x62
github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.StreamServerInterceptor.func1({0x5ff7300, 0xc00427e000}, {0x770aeb8?, 0xc0007acb70?}, 0xa93a4b0, 0xc00bb847e0)
  github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:164 +0x6c4
github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run.func1({0x5ff7300?, 0xc00427e000?}, {0x770aeb8?, 0xc0007acb70?})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:982 +0x5b
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x5ff7300, 0xc00427e000}, {0x770aeb8, 0xc0007acb70}, 0xc0007acb40?, 0xc00d1d2a80)
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:284 +0x74
github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run.func1({0x5ff7300?, 0xc00427e000?}, {0x770aeb8?, 0xc0007acb70?})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:985 +0x83
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc000a16000?, {{0x0?}, {0x770ae28?, 0xc00d5c83f0?}}}, {0x5ff7300, 0xc00427e000}, {0x770aeb8, 0xc0007acb70}, 0xa93a4b0, 0xc00d1d2a80)
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:167 +0x3f7
github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run.func1({0x5ff7300?, 0xc00427e000?}, {0x770aeb8?, 0xc0007acb70?})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:985 +0x83
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x8?, 0x0?})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:251 +0x2d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00b62a800, {0x76d6b60, 0xc0007acb40}, {0xc00526aec8?, 0x4631ff?}, 0xc00526ae80)
  github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:322 +0xd1
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x5ff7300, 0xc00427e000}, {0x770aeb8?, 0xc0007acb70?}, 0xa93a4b0, 0xc00d1d2a80)
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:250 +0xef
github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run.func1({0x5ff7300?, 0xc00427e000?}, {0x770aeb8?, 0xc0007acb70?})
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:985 +0x83
github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run({0xc00d630b60, 0x4, 0x4}, {0x5ff7300, 0xc00427e000}, {0x770aeb8, 0xc0007acb70}, 0xa93a4b0, 0xc00bb847e0)
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:987 +0x142
github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.MuxRangeFeed.func3()
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1314 +0xfe
created by github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.MuxRangeFeed
  github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1299 +0x3aa


goroutine 16905 [chan receive, 14 minutes]:
testing.(*T).Run(0xc00d61c1a0, {0x610d983?, 0x609dd38?}, 0xc009b9f650)
  GOROOT/src/testing/testing.go:1494 +0x37a
github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamproducer_test.TestStreamPartition(0xc00d61c1a0)
  github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamproducer_test/pkg/ccl/streamingccl/streamproducer/replication_stream_test.go:358 +0x414
testing.tRunner(0xc00d61c1a0, 0x645ee28)
  GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
  GOROOT/src/testing/testing.go:1493 +0x35f
```

cc @stevendanna for triage

--- a/pkg/ccl/streamingccl/streamproducer/event_stream.go
+++ b/pkg/ccl/streamingccl/streamproducer/event_stream.go
@@ -308,6 +308,8 @@ func (s *eventStream) flushEvent(ctx context.Context, event *streampb.StreamEven
 		return ctx.Err()
 	case s.streamCh <- tree.Datums{tree.NewDBytes(tree.DBytes(data))}:
 		return nil
+	case <-s.doneChan:
+		return nil
 	}
 }
 

